<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    <script>
        /*
            catch 有返回值，它的返回值参照promise的then方法
        */

        // fn()
        async function fn() {

            try {
                await new Promise((resolve, reject) => {
                    setTimeout(() => {
                        console.log('1')
                        // resolve('1')
                        reject('错误')
                    }, 500)
                })

                new Promise((resolve, reject) => {
                    setTimeout(() => {
                        console.log('2')
                        resolve('2')
                    }, 1500)
                })

                await new Promise((resolve, reject) => {
                    setTimeout(() => {
                        console.log('3')
                        resolve('3')
                    }, 1000)
                })
            }catch(error){
                console.log(error)
            }
            
        }


        // test
        // console.log(a)   // 报错,并且会阻止程序执行

        try{
            console.log(a)
        }catch(error) {
            console.log(error)   // 打印出错误信息，不会阻止程序执行
        }
        

    </script>
</body>
</html>